src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,311 @@
+
+    VerifyOrReturn(delegate != nullptr, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure));
+
+    delegate->GetCurrentOperationalState(opState);
+
+    if (opState.operationalStateID != to_underlying(OperationalStateEnum::kRunning))
+    {
+        delegate->HandleStartStateCallback(err);
+    }
+    response.commandResponseState = err;
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void OperationalStateServer::HandleStopState(HandlerContext & ctx, const Commands::Stop::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandleStopState");
+    Commands::OperationalCommandResponse::Type response;
+    Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    GenericOperationalState opState;
+
+    VerifyOrReturn(delegate != nullptr, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure));
+
+    delegate->GetCurrentOperationalState(opState);
+
+    if (opState.operationalStateID != to_underlying(OperationalStateEnum::kStopped))
+    {
+        delegate->HandleStopStateCallback(err);
+    }
+    response.commandResponseState = err;
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+template <typename RequestT, typename FuncT>
+void OperationalStateServer::HandleCommand(HandlerContext & handlerContext, FuncT func)
+{
+    if (!handlerContext.mCommandHandled && (handlerContext.mRequestPath.mCommandId == RequestT::GetCommandId()))
+    {
+        RequestT requestPayload;
+
+        //
+        // If the command matches what the caller is looking for, let's mark this as being handled
+        // even if errors happen after this. This ensures that we don't execute any fall-back strategies
+        // to handle this command since at this point, the caller is taking responsibility for handling
+        // the command in its entirety, warts and all.
+        //
+        handlerContext.SetCommandHandled();
+
+        if (DataModel::Decode(handlerContext.mPayload, requestPayload) != CHIP_NO_ERROR)
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath,
+                                                     Protocols::InteractionModel::Status::InvalidCommand);
+            return;
+        }
+
+        func(handlerContext, requestPayload);
+    }
+}
+
+CHIP_ERROR OperationalStateServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (aPath.mAttributeId)
+    {
+    case OperationalState::Attributes::OperationalStateList::Id: {
+        Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+        GenericOperationalState opState;
+        size_t index = 0;
+
+        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(NotSpecified, "Delegate is nullptr"));
```

Comment:
```suggestion
        VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is nullptr"));
```
and same for all the other NotSpecified.